src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -243,50 +253,59 @@ class ThetaStarPathPlanner : public PathPlanner
      *
      * @return cell in grid
      */
-    Coordinate pointToCoordinate(Point p);
+    Coordinate convertPointToCoord(const Point &p) const;
 
     /**
-     * Try to find a path to destination and leave
+     * Try to find a path to end and leave
      * trail markers along the way
      *
-     * @param dest_coord destination coordinates
+     * @param end_coord end coordinates
      *
-     * @return if path to destination was found
+     * @return if path to end was found
      */
-    bool findPathToDestination(Coordinate dest_coord);
+    bool findPathToEnd(const Coordinate &end_coord);
 
     /**
-     * Check for invalid or blocked src_coord and dest_coord and
+     * Update vertex for all successors of current_coord
+     *
+     * @param current_coord The current coordinate
+     * @param end_coord end coordinates
+     *
+     * @return if path to end was found among successors
+     */
+    bool visitSuccessors(const Coordinate &current_coord, const Coordinate &end_coord);
+
+    /**
+     * Check for non-navigable or blocked start_coord and end_coord and
      * adjust parameters accordingly
      *
-     * @param src_coord source coordinate
-     * @param dest_coord destination coordinate
+     * @param [in/out] start_coord source coordinate
+     * @param [in/out] end_coord end coordinate
      *
-     * @return true if there is no path to destination
+     * @return true if there is no path to end
      */
-    bool checkForInvalidOrBlockedCases(Coordinate &src_coord, Coordinate &dest_coord);
+    bool nonNavigableOrBlockedCases(Coordinate &start_coord, Coordinate &end_coord);
 
     /**
-     * Check if start to destination is at least a grid cell away
+     * Check if start to end is at least a grid cell away
      *
      * @param start start point
-     * @param destination destination point
+     * @param end end point
      *
-     * @return true start to destination is within threshold
+     * @return true start to end is within threshold
      */
-    bool isStartToDestinationWithinThreshold(const Point &start,
-                                             const Point &destination);
+    bool isStartToEndWithinThreshold(const Point &start, const Point &end) const;
 
     /**
-     * Check if start to closest destination is at least a grid cell away
+     * Check if start to closest end is at least a grid cell away
```

Comment:
Can there be more than one end point? Confused what the difference is between this and the above function (haven't read implementation yet)

Reply:
if the end point is blocked, then the closest end is the closest unblocked point. It has to do with the oscillation stuff, which I don't want to deal with in this PR

Reply:
created #1448